docs/user-guide.md

```diff
@@ -0,0 +1,856 @@
+        </script>
+    </body>
+</html>
+```
+
+PolyScript's capabilities, upon which PyScript is built, can be summarised as:
+
+* Evaluation of code via [`<script>` tags](https://pyscript.github.io/polyscript/#how-scripts-work).
+* Handling of
+  [browser events](https://pyscript.github.io/polyscript/#how-events-work)
+  via code evaluated by an interpreter supported by PolyScript.
+* A [clear way to use workers](https://pyscript.github.io/polyscript/#xworker)
+  via the `XWorker` class and its related reference, `xworker`.
+* [Custom scripts](https://pyscript.github.io/polyscript/#custom-scripts) to
+  enrich PolyScript's capabilities.
+* A [ready event](https://pyscript.github.io/polyscript/#ready-event)
+  dispatched when an interpreter is ready and about to run code.
+* [Multipe interpreters](https://pyscript.github.io/polyscript/#interpreter-features)
+  (in addition to Pyodide and MicroPython, PolyScript works with Lua and Ruby -
+  although these are beyond the scope of this project).
+
+### The stack
+
+The stack describes how the different building blocks _inside_ a PyScript
+application relate to each other:
+
+<img src="../assets/images/platform.png"/>
+
+* Everything happens inside the context of the browser (represented by the
+  black border). It means the browser tab for your PyScript app is your
+  sandboxed computing environment.
+
+!!! failure
+
+    Python solely running in the browser is an unfamiliar concept that some
+    fail to remember.
+
+    * PyScript isn't running on a server hosted in the cloud.
+    * PyScript doesn't use the version of Python running natively on the user's
+      operating system.
+    * **PyScript only runs IN THE BROWSER (and nowhere else).**
+
+* At the bottom of the stack are the Python interpreters compiled to WASM. They
+  evaluate your code and interact with the browser via the [FFI](#ffi).
+* The PyScript layer makes it easy to use and configure the Python
+  interpreters. There are two parts to this:
+    1. The PolyScript kernel (see above), that bootstraps everything and
+       provides the core capabilities.
+    2. PyScript and related plugins that sit atop PolyScript to give us an
+       easy-to-use Python platform _in the browser_.
+* Above the PyScript layer are either:
+    1. Application frameworks, modules and libraries written in Python that you
+       use to create useful applications.
+    2. Your code (that's your responsibility).
+
+## Lifecycle
+
+If the architecture explains how components relate to each other, the lifecycle
+explains how things unfold. It's important to understand both: it will
+help you think about your own code and how it sits within PyScript.
+
+Here's how PyScript unfolds through time:
+
+* The browser is directed to a URL. The response is HTML.
+* As part of parsing the HTML response the browser encounters the `<script>`
+  tag that references PyScript. PyScript is loaded and evaluated as a module,
```

Comment:
```suggestion
  tag that references PyScript. PyScript is loaded and evaluated as a JS module,
```

although probably the most correct technical term is `ES Module` (for EcmaScript Module, I think?).
Probably @WebReflection knows better

Reply:
"JavaScript" module..?